### PR TITLE
Refactor/removemodel

### PR DESCRIPTION
**Pull request by kddejong**
## Proposed changes

Sorry, but lots of changes here.  Trying to get agreement on how to put the new account services and data modeling into place.  This removes the model package and adds an account service.   I think this handles most of the feedback that has been received on how to make this better.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (changes to code, which do not change application behavior)

## Checklist

- [ ] I have filled out this PR template
- [ ] I have read the [CONTRIBUTING](https://github.com/Optum/dce/blob/master/docs/CONTRIBUTING.md) doc
- [ ] I have added automated tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (`README.md`, inline comments, etc.)
- [ ] I have updated the `CHANGELOG.md` under a `## next` release, with a short summary of my changes

### CODE REVIEW

**Comment by kddejong** [comment]
Tests are failing today because of #184.  Will re-run tomorrow.

File: pkg/account/model.go [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+
+import (
+	"github.com/Optum/dce/pkg/errors"
+	validation "github.com/go-ozzo/ozzo-validation"
+)
 
 // Account - Handles importing and exporting Accounts and non-exported Properties
 type Account struct {
 	ID                  *string                `json:"id,omitempty" dynamodbav:"Id"`                                             // AWS Account ID
-	Status              *AccountStatus         `json:"accountStatus,omitempty" dynamodbav:"AccountStatus,omitempty"`             // Status of the AWS Account
```
Comment (kddejong) [inline pkg/account/model.go]: Now that this is apart of the account package removing stuttering.

File: pkg/account/service.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+	Publish() error
+}
+
+// Manager manages all the actions against an account
+type Manager interface {
+	Setup(adminRole string) error
+}
+
+// Service is a type corresponding to a Account table record
+type Service struct {
```
Comment (kddejong) [inline pkg/account/service.go]: This is the new Account Service.  This separates the data of the Account and helps provide a generic Account Servicer.

File: pkg/data/lease.go [hunk 3]
```diff
@@ -4,20 +4,27 @@ import (
...
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/dynamodb"
 	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
 	"github.com/aws/aws-sdk-go/service/dynamodb/expression"
 )
 
-// WriteLease the Lease record in DynamoDB
+// Lease - Data Layer Struct
+type Lease struct {
```
Comment (kddejong) [inline pkg/data/lease.go]: Need to have the Lease stuff separate.  I think we are on a path to have a generic Data service but need to figure out how to handle some of the inconsistencies between the tables first.

File: pkg/config/config.go [hunk 4]
```diff
@@ -256,3 +257,14 @@ func getKeyPtrs(aMap map[string]ParameterStoreVal) []*string {
 	}
 	return keys
 }
+
+// AccountSvc returns the account Service for you
+func (config *ConfigurationBuilder) AccountSvc() accountiface.Servicer {
+
+	var accountSvc accountiface.Servicer
+	if err := config.GetService(&accountSvc); err != nil {
+		panic(err)
```
Comment (kddejong) [inline pkg/config/config.go]: Left this as a Panic so the dot notation could be used.  Instead doing (accountiface.Servicer, error) we can just return the appropriate service allowing for someone to do `config.AccountSvc().Update(...)`

File: pkg/account/accountiface/servicer.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
+//
+
+package accountiface
+
+import (
+	"github.com/Optum/dce/pkg/account"
+)
+
+// Servicer makes working with the Account Service struct easier
+type Servicer interface {
```
Comment (joshmarsh) [inline pkg/account/accountiface/servicer.go]: This may read better as `AccountServicer`.

**Comment by kddejong** [comment]
This would replace #211

**Pull request merged**